gateway-ha/src/main/java/io/trino/gateway/ha/handler/ProxyUtils.java

```diff
@@ -50,7 +50,14 @@ public final class ProxyUtils
      * capitalization.
      */
     private static final Pattern QUERY_ID_PARAM_PATTERN = Pattern.compile(".*(?:%2F|(?i)query_?id(?-i)=|^)(\\d+_\\d+_\\d+_\\w+).*");
-    private static final Pattern EXTRACT_BETWEEN_SINGLE_QUOTES = Pattern.compile("'([^\\s']+)'");
+
+    /**
+     * This regular expression extracts the query id from a CALL system.runtime.kill_query procedure call. It extracts the first string between quotes
+     * following the open parentheses after system.runtime.kill_query. The pattern handles the optional named arguments, and optional message argument
+     * as well as arbitrary whitespace.
+     */
+    private static final Pattern KILL_QUERY_PROCEDURE_PATTERN
+            = Pattern.compile(".*system\\.runtime\\.kill_query\\s*\\(\\s*(query_id\\s*=>)?\\s*'([^\\\\s]+?)'(,\\s*(message\\s*=>\\s*)?('.*'))?\\)");
```

Comment:
The pattern looks wrong. Users can run the procedure without `system.runtime.` if the default catalog and schema are set.

Reply:
the unqualified case is handled now as well, ptal. I think this should cover everything, but if not I am considering using the parser instead of this regex